pyomo/contrib/doe/measurements.py

```diff
@@ -25,252 +25,48 @@
-        For example, with a partial differential algebraic equation model, 
-        these measurement index sets can specify which spatial and temporal coordinates each measurement is available. 
-        Moreover, this class supports defining the covariance matrix for all measurements.
-
-        Parameters
-        ----------
-        measurement_index_time:
-            a ``dict``, keys are measurement variable names, 
-                if there are extra index, for e.g., Var[scenario, extra_index, time]:
-                    values are a dictionary, keys are its extra index, values are its measuring time points. 
-                if there are no extra index, for e.g., Var[scenario, time]:
-                    values are a list of measuring time point.
-            For e.g., for the kinetics illustrative example, it should be {'C':{'CA':[0,1,..], 'CB':[0,2,...]}, 'k':[0,4,..]},
-            so the measurements are C[scenario, 'CA', 0]..., k[scenario, 0]....
-        variance:
-            a ``dict``, keys are measurement variable names, values are a dictionary, keys are its extra index,
-            values are its variance (a scalar number), values are its variance if there is no extra index for this measurement.
-            For e.g., for the kinetics illustrative example, it should be {'C':{'CA': 10, 'CB': 1, 'CC': 2}}.
-            If given None, the default is {'C':{'CA': 1, 'CB': 1, 'CC': 1}}.
-        ind_string:
-            a ''string'', used to flatten the name of variables and extra index. Default is '_index_'.
-            For e.g., for {'C':{'CA': 10, 'CB': 1, 'CC': 2}}, the reformulated name is 'C_index_CA'.
         """
-        self.measurement_all_info = measurement_index_time
-        self.ind_string = ind_string
-        # a list of measurement names
-        self.measurement_name = list(measurement_index_time.keys())
-        # begin flatten
-        self._name_and_index_generator(self.measurement_all_info)
-        self._generate_flatten_name(self.name_and_index)
-        self._generate_variance(self.flatten_measure_name, variance, self.name_and_index)
-        self._generate_flatten_timeset(self.measurement_all_info, self.flatten_measure_name, self.name_and_index)
-        self._model_measure_name()
+        super().__init__()
+        self._generate_variance(variance)
 
-        # generate the overall measurement time points set, including the measurement time for all measurements
-        flatten_timepoint = list(self.flatten_measure_timeset.values())
-        overall_time = []
-        for i in flatten_timepoint:
-            overall_time += i
-            timepoint_overall_set = list(set(overall_time))
-        self.timepoint_overall_set = timepoint_overall_set
+    def specify(self, self_define_res):
 
+        self.measurement_name = super().specify(self_define_res)
 
-    def _name_and_index_generator(self, all_info):
+    def add_elements(self, var_name, extra_index=None, time_index=[0]):
```

Comment:
Why is add_elements defined here? Why not just inherit the version from SpecialSet?

Reply:
using super()